### DISCUSSION TITLE
Firebase Functions

### DISCUSSION THREAD

**Discussion by Dellybro** [invertase/react-native-firebase#5353] [Q&A]
Hello,

When calling a function using httpscallable() wrapped inside a try catch, my catch is never called.

On the server I am using the following code.

```
return new functions.https.HttpsError('invalid-argument', error.message, {
      message: error.message,
});
``` 

On the client, catch is never called and the data returned is in the `response.data` object.

How can i utilize my catch rather then having to determine the success based on whether or not httpErrorCode exists in my response.data?
```
try {
  const {data} = await functions().httpsCallable('someFn')()
   if (data.httpErrorCode) throw new Error(data.details.message);
} catch (error) {
   // httpsCallable does not throw an error when returning new https.HttpsError.
}
```

**Comment by mikehardy** [comment]
How does firebase-js-sdk behave when you execute that code? I think if the function executes "successfully" (returns its own error code perhaps but does not fail for some reason outside of your function code) you have to interpret your own error message as you indicate

I believe if it fails for one of these reasons https://firebase.google.com/docs/reference/js/firebase.functions#functionserrorcode - that's when you will go to your catch block

in other words I believe there is a separation between "functions in general failed" (goes to catch block) and "your function wants to return an error" (you parse that in try using your own code)

But if we are deviating from firebase-js-sdk in some way here we can certainly harmonize, please let us know your findings

**Reply by Dellybro** [reply]
Hey Mike appreciate the reply

So in my experience, when an API returns anything other then 200-300 response code usually packages such as Fetch and Axios fail into the catch block. So i was thinking by returning `return new functions.https.HttpsError('invalid-argument' ...)` this would behave the same.

The link you sent me, https://firebase.google.com/docs/reference/js/firebase.functions#functionserrorcode, shows one of the error codes which i am using "invalid-argument" but doesn't necessarily fall into the catch block.

It seems as though returning the http error in the functions generates a normalized json for this error but still returns a 200.

So i guess the question is, when using `functions.https.onCall`, in a google cloud function, and returning `functions.https.HttpsError` does this also return a 200, thus never activating my catch block.

I think if this is the case, then it would make sense that the SDK does not fail over in the catch, but if returning `functions.https.HttpsError` does indeed send back a 400 response code, the SDK maybe behaving inadequately.

**Reply by mikehardy** [reply]
@Dellybro 

> How does firebase-js-sdk behave when you execute that code?

> But if we are deviating from firebase-js-sdk in some way here we can certainly harmonize, please let us know your findings
